code/gateway/dataaccess.q

```diff
@@ -173,29 +209,73 @@ partdict:{[input]
 // function to adjust the queries being sent to processes to prevent overlap of
 // time clause and data being queried on more than one process
 adjustqueries:{[options;part]
-    // if only one process then no need to adjust
-    if[2>count p:options`procs;:options];
+    // if only one process and not striped then no need to adjust
+    if[(not b:6h~type raze key part)&2>count p:options`procs;:options];
+
+    overlap:max{x~/:key y}[;part]each p;
+    part:key[part][where overlap]!value[part]where overlap;
+
+    // no overlaps between procs and starttime and endtime
+    if[0=count part;
+        '`$.checkinputs.formatstring[.schema.errors[`timeprocoverlap;`errormessage];`procs`starttime`endtime!(p;options`starttime;options`endtime)]];
+
     // get the date casting where relevant
     st:$[a:-14h~tp:type start:options`starttime;start;`date$start];
     et:$[a;options`endtime;`date$options`endtime];
+
     // get the dates that are required by each process
-    dates:group key[part]where each{within[y;]each value x}[part]'[l:st+til 1+et-st];
+    dates:$[b;
+        {key[y]!where each flip x};
+        {group key[y]where each x}][;part]{within[y;]each value x}[part]'[l:st+til 1+et-st];
```

Comment:
If (b) any serverid by servertype is striped: part dict is keyed by a unique (nested) list of serverids rather than servertype so that other part of the code will not work in this instance.
For e.g. the key is (1i;2i;3i;4 5i) vs `rdb`rdb`hdb. Let me know if that makes sense!

Reply:
OK. Is there a reason we have to keep using servertype for the non-striped case or can we switch that to serverid as well? Will require a bit more testing to make sure nothing breaks but I think that should hopefully be covered by existing unit tests.

My thinking here is to reduce complexity & "special handling" wherever possible - if we can use the same code in all cases, keeps thinks simpler.

Reply:
Yes I think it makes sense to switch everything to the same format but I was trying to retain the original functionality previously. In that case, I will try to convert all to serverid instead so we don't require any "special handling".